drivers/sensor/adi/max32664c/max32664c.c

```diff
@@ -0,0 +1,1122 @@
+		return -EINVAL;
+	}
+
+	data->op_mode = MAX32664C_OP_MODE_IDLE;
+
+	return 0;
+}
+
+static int max32664c_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct max32664c_data *data = dev->data;
+
+	switch (data->op_mode) {
+	case MAX32664C_OP_MODE_STOP_ALGO:
+	case MAX32664C_OP_MODE_IDLE:
+		LOG_DBG("Device is idle, no data to fetch!");
+		return -EAGAIN;
+	case MAX32664C_OP_MODE_SCD:
+		k_msgq_get(&data->scd_report_queue, &data->scd, K_NO_WAIT);
+		break;
+#ifdef CONFIG_MAX32664C_USE_EXTENDED_REPORTS
+	case MAX32664C_OP_MODE_ALGO_AEC_EXT:
+	case MAX32664C_OP_MODE_ALGO_AGC_EXT:
+		k_msgq_get(&data->ext_report_queue, &data->ext, K_NO_WAIT);
+#else
+	case MAX32664C_OP_MODE_ALGO_AEC:
+	case MAX32664C_OP_MODE_ALGO_AGC:
+		k_msgq_get(&data->report_queue, &data->report, K_NO_WAIT);
+#endif /* CONFIG_MAX32664C_USE_EXTENDED_REPORTS */
+	/* Raw data are reported with normal and extended algorithms so we need to fetch them too */
+	case MAX32664C_OP_MODE_RAW:
+		k_msgq_get(&data->raw_report_queue, &data->raw, K_NO_WAIT);
+		break;
+	default:
+		return -EINVAL;
+	};
+
+	return 0;
+}
+
+static int max32664c_channel_get(const struct device *dev, enum sensor_channel chan,
+				 struct sensor_value *val)
+{
+	struct max32664c_data *data = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_X: {
+		val->val1 = data->raw.acc.x;
+		break;
+	}
+	case SENSOR_CHAN_ACCEL_Y: {
+		val->val1 = data->raw.acc.y;
+		break;
+	}
+	case SENSOR_CHAN_ACCEL_Z: {
+		val->val1 = data->raw.acc.z;
+		break;
+	}
+	case SENSOR_CHAN_GREEN: {
+		val->val1 = data->raw.PPG1;
+		break;
+	}
+	case SENSOR_CHAN_IR: {
+		val->val1 = data->raw.PPG2;
+		break;
+	}
+	case SENSOR_CHAN_RED: {
+		val->val1 = data->raw.PPG3;
+		break;
+	}
+	case SENSOR_CHAN_HEARTRATE: {
+		val->val1 = data->report.hr;
+		val->val2 = data->report.hr_confidence;
+		break;
+	}
+	case SENSOR_CHAN_RESPIRATION_RATE: {
+		val->val1 = data->report.rr;
+		val->val2 = data->report.rr_confidence;
+		break;
+	}
+	case SENSOR_CHAN_BLOOD_OXYGEN_SATURATION: {
+		val->val1 = data->report.spo2;
+		val->val2 = data->report.spo2_confidence;
+		break;
+	}
+	case SENSOR_CHAN_SKIN_CONTACT: {
+		val->val1 = data->report.scd_state;
+		break;
+	}
+	default: {
+		LOG_ERR("Channel %u not supported!", chan);
+		return -EINVAL;
+	}
+	}
+
+	return 0;
+}
+
+static int max32664c_attr_set(const struct device *dev, enum sensor_channel chan,
+			      enum sensor_attribute attr, const struct sensor_value *val)
+{
+	int err;
+	uint8_t tx[5];
+	uint8_t rx;
+	struct max32664c_data *data = dev->data;
+
+	if ((chan != SENSOR_CHAN_HEARTRATE) && (chan != SENSOR_CHAN_BLOOD_OXYGEN_SATURATION) &&
+	    (chan != SENSOR_CHAN_RESPIRATION_RATE) && (chan != SENSOR_CHAN_SKIN_CONTACT) &&
+	    (chan != SENSOR_CHAN_ACCEL_X) && (chan != SENSOR_CHAN_ACCEL_Z) &&
+	    (chan != SENSOR_CHAN_ACCEL_Y) && (chan != SENSOR_CHAN_RED) &&
+	    (chan != SENSOR_CHAN_GREEN) && (chan != SENSOR_CHAN_IR) && (chan != SENSOR_CHAN_ALL)) {
```

Comment:
CI is failing because not all of these sensor channels are defined in the enumeration. Suggest using custom channels - search for `SENSOR_CHAN_PRIV_START` for some examples.
https://github.com/zephyrproject-rtos/zephyr/actions/runs/16655993400/job/47141311165?pr=93674#step:13:1112

Reply:
Hi @MaureenHelm,
I´ve created PR #93484 to add the missing channels and I would like to wait with this PR until the other PR is merged. IMO it doesn´t make sense to create private channels for this driver because they are probably needed by other devices.

Reply:
@MaureenHelm alright. So shall we keep them as private for now or shall I move them from #93484 as regular channels/attributes? IMO

 - SENSOR_ATTR_OP_MODE -> Should be added as a regular attribute because it can be used by all types of sensors to switch between modes
 - SENSOR_CHAN_STEP_COUNTER -> Should be added as a regular channel because it can be used by IMUs for step counting
 - SENSOR_CHAN_ACTIVITY -> Same here
 
 The rest is (AFAIK) specific to my sensor and can be left private for now